lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -529,6 +529,31 @@ namespace Js
                 VirtualTableInfo<JavascriptNativeFloatArray>::HasVirtualTable(aValue));
     }
 
+    bool JavascriptArray::IsInlineSegment(SparseArraySegmentBase *seg, JavascriptArray *pArr)
+    {
+        SparseArraySegmentBase* inlineHeadSegment = nullptr;
+        if (JavascriptNativeArray::Is(pArr))
+        {
+            if (JavascriptNativeFloatArray::Is(pArr))
+            {
+                inlineHeadSegment = DetermineInlineHeadSegmentPointer<JavascriptNativeFloatArray, 0, true>((JavascriptNativeFloatArray*)pArr);
+            }
+            else if (JavascriptNativeIntArray::Is(pArr))
```

Comment:
>if (JavascriptNativeIntArray::Is(pArr) [](start = 17, length = 38)

This can be converted to an assert instead as JavascriptNativeArray::Is already verified that it is either a NativeFloat or NativeInt array